presto-tests/src/test/java/com/facebook/presto/tests/TestSqlFunctions.java

```diff
@@ -120,6 +146,19 @@ public void testCreateFunction()
         assertQuerySucceeds("CREATE FUNCTION TESTING.TEST.TAN (x int) RETURNS double RETURN sin(x) / cos(x)");
         assertQuerySucceeds("CREATE FUNCTION testing.test.tan (x double) RETURNS double LANGUAGE JAVA RETURN sin(x) / cos(x)");
 
+        // function with enums
+        assertQuerySucceeds("CREATE FUNCTION testing.test.is_china(country testing.enum.country) RETURNS boolean RETURN country = testing.enum.country.CHINA");
+        assertQuery("SELECT testing.test.is_china(testing.enum.country.CHINA)", "SELECT true");
+        assertQuery("SELECT testing.test.is_china(testing.enum.country.\"भारत\")", "SELECT false");
+
+        assertQuerySucceeds("CREATE FUNCTION testing.test.get_mood(x varchar) RETURNS testing.enum.mood RETURN if(x='foo', testing.enum.mood.happy, testing.enum.mood.curious)");
+        MaterializedResult rows = computeActual("SELECT testing.test.get_mood('foo')");
+        assertEquals(rows.getTypes().get(0).getDisplayName(), "testing.enum.mood");
+        assertEquals(rows.getMaterializedRows().get(0).getFields().get(0), 0L);
+        rows = computeActual("SELECT testing.test.get_mood('bar')");
+        assertEquals(rows.getTypes().get(0).getDisplayName(), "testing.enum.mood");
+        assertEquals(rows.getMaterializedRows().get(0).getFields().get(0), -2L);
```

Comment:
Can we add some failure test cases. e.g. referring a enum value that does not exists `testing.enum.country.C`.